src/components/autocomplete/js/autocompleteController.js

```diff
@@ -196,6 +197,16 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
   }
 
   /**
+   * Moves the live region within the autocomplete to ensure the live region is
+   * is not hidden from assistive technologies when the autocomplete appears
+   * within a dialog.
+   */
+  function moveLiveRegion () {
+    angular.element($mdLiveAnnouncer._liveElement).detach();
+    elements.$.main.append($mdLiveAnnouncer._liveElement)
```

Comment:
The `$mdLiveAnnouncer` is a global singleton; it does not belong to the autocomplete. With this, each autocomplete will "steal" the aria-live element when it's created. Said element will also be destroyed if whichever autocomplete stole it most recently is destroyed, breaking all subsequent announcing. 

The easiest way to fix this issue is probably to have the dialog specifically ignore the aria-live element when marking everything as `aria-hidden`.